uplink/piecestore/download.go

```diff
@@ -18,10 +18,12 @@ import (
 )
 
 // Downloader is interface that can be used for downloading content.
-// It matches signature of `io.ReadCloser`.
+// It matches signature of `io.ReadCloser`, with one extra function,
+// GetHashAndLimit(), used for accessing information during GET_REPAIR
 type Downloader interface {
```

Comment:
maybe if this is no longer implementing `io.ReadCloser`, maybe a new interface could be created that wraps the Downloader interface with the additional `GetHashAndLimit` method? I'm not sure if this would solve the problem you were talking about though.

Reply:
I don't think that would help unfortunately because `GetHashAndLimit` needs to be on the downloader since that is where the data is received from.